SSLP_analyzer/SSLP_App/views.py

```diff
@@ -0,0 +1,306 @@
+                population_fromfile = populations[0]
+                messages.info(request, f"Population found in file is not currently in database, the default value {population_fromfile} was used instead.")
+            combinations = {}
+            request.session["combinations"] = {}
+
+            for line in input_data_list[1:-1]:
+                items = line.split(';')[:-1]
+                id = items[0]
+                sslps = [int(x) for x in items[1:]]
+                combinations[id] = {'Population': population_fromfile,
+                                    'SSLPS': sslps}
+            request.session["combinations"] = combinations
+            if table_haplotype_filled != 1:
+                haplotype_table = table_haplotype_filled
+                total_perc = f'{total_perc_int:.1f}%'
+                title = f'{SSLPs} {population_fromfile}'
+            else:
+                title = "Current selection does not return results"
+                switch_title = True
+                total_perc = ""
+        except IndexError:
+            messages.warning(request, 'File cannot be uploaded due to the wrong format.')
+        except ValueError:
+            messages.warning(request, 'One or more lines of file include text where numbers are expected.')
+    return render(request, 'homepage.html', {
+        'Title': title,
+        'data': haplotype_table,
+        'saved_results': get_saved_results(request),
+        'chrom_lengths': all_sslps,
+        'populations': populations,
+        'likelihood': total_perc,
+        'switch_title': switch_title,
+        'result_check': "" if not table_haplotype_filled or table_haplotype_filled == 1 else True 
+    })
+
+
+def list_of_sslps():
+    """
+        Summary:
+            This function reads a JSON file that contains SSLPs and populations data. It extracts the SSLPs and populations
+            from the file and returns them as separate lists.
+        Returns:
+            Two lists: all_sslps and populations.
+    """
+    file_path = os.path.join(
+        os.path.dirname(os.path.dirname(__file__)), "haplotypes.json")
+    all_sslps, populations = [], []
+    with open(file_path, "r") as file:
+        haplotype_file = json.load(file)
+        populations = list(haplotype_file.keys())
+        for _, pop in haplotype_file.items():
+            for _, sslps in pop.items():
+                all_sslps.extend((sslps))
+    all_sslps = sorted(list(set(all_sslps)))
+    return all_sslps, populations
+
+
+def get_saved_results(request):
+    """
+        Returns:
+            dict: The combinations dictionary containing the saved results.
+        Summary:
+            This function takes the combinations dictionary from the session, which contains the saved results. If the
+            combinations dictionary is not empty, it returns the dictionary.
+    """
+    combinations = request.session.get('combinations', {})
+    if combinations != {}:
+        return combinations
```

Comment:
If this `if` is not true, the function won't return anything. I assume in Python a `void` return value will make the code return `None`. Can the code that calls this function handle that?

Reply:
The output of this function is used by the template renderer. If it returns None it will not show any results. Which is the expected behavior.